src/fabric/src/fabric2_fdb.erl

```diff
@@ -72,24 +72,29 @@
 
 
 transactional(Fun) ->
-    do_transaction(Fun, undefined).
-
+    BeforeTx = start_span_fun(Fun, [], #{}),
```

Comment:
I tried using with_span it doesn't work as needed. Because we want to trace do_transaction and we don't have database name in there. We want to trace do_transaction because we want to see retries in the trace.

Reply:
I don't think I agree with the premise that we specifically want to trace `do_transaction`. Using `with_span` further up in `fabric2_fdb:transactional` gives us an easy place to track the database name and covers the scope of the entire transaction. I do understand wanting to also trace transaction retries, however that's not actually happening because `BeginTx` and `AfterTx` are outside of the closure passed to `erlfdb:transactional/2`.

To track retries I'd add another span around [1] when we actually invoke the closure. And if we update `ctrace:with_span` to capture exceptions like `passage_pd` does we have the extra benefit of seeing the `reopen` exceptions as well as capturing any FoundationDB error codes in the second span.

[1] https://github.com/apache/couchdb/blob/03f3054e5987bc0c1927bc417c41f3a518391f6b/src/fabric/src/fabric2_fdb.erl#L1376